core/src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -1,10 +1,27 @@
 package pureconfig
 
+/**
+ * A mapping between case class fields and their respective keys in the config.
+ */
 trait ConfigFieldMapping extends (String => String) {
+
   def apply(fieldName: String): String
+
+  /**
+   * Returns a `ConfigFieldMapping` that uses this mapping with some exceptions.
+   *
+   * @param exceptions the exceptions for this mapping as pairs (field, configKey)
```

Comment:
I think the distinction between `ConfigFieldMapping` and a `ProductHint` provides that information. A `ConfigFieldMapping` has no type parameter and has no effect as an implicit. On the other hand, a `ProductHint` has a type parameter restricting the hint to certain types (possibly one or all). Different product hints may be configured with different mappings, so "any override will be applied globally throughout the configuration" isn't entirely accurate. Do you still think it would be better to improve the documentation?